tests/unittests/dataset_tests.py

```diff
@@ -711,6 +711,233 @@ def test_index_in_progress(self):
+                {
+                    fo.DynamicEmbeddedDocument(label="foo", count=2),
+                    fo.DynamicEmbeddedDocument(label="bar", count=1),
+                },
+                None,
+            ],
+        )
+        self.assertListEqual(
+            to_sets(dataset.values("frames_gt_confidence_by_label")),
+            [
+                {
+                    fo.DynamicEmbeddedDocument(label="foo", min=0.1, max=0.9),
+                    fo.DynamicEmbeddedDocument(label="bar", min=0.5, max=0.5),
+                },
+                None,
+            ],
+        )
+
+        summary_fields = dataset.list_summary_fields()
+
+        self.assertTrue(dataset.get_field("gt_label").read_only)
+        self.assertTrue(dataset.get_field("gt_confidence").read_only)
+        self.assertFalse(dataset.get_field("gt_label_counts").read_only)
+        self.assertFalse(dataset.get_field("gt_confidence_by_label").read_only)
+
+        self.assertTrue(dataset.get_field("frames_gt_label").read_only)
+        self.assertTrue(dataset.get_field("frames_gt_confidence").read_only)
+        self.assertFalse(dataset.get_field("frames_gt_label_counts").read_only)
+        self.assertFalse(
+            dataset.get_field("frames_gt_confidence_by_label").read_only
+        )
+
+        self.assertSetEqual(
+            set(summary_fields),
+            {
+                "gt_label",
+                "gt_confidence",
+                "gt_label_counts",
+                "gt_confidence_by_label",
+                "frames_gt_label",
+                "frames_gt_confidence",
+                "frames_gt_label_counts",
+                "frames_gt_confidence_by_label",
+            },
+        )
+
+        db_indexes = dataset.list_indexes()
+
+        self.assertTrue("gt_label" in db_indexes)
+        self.assertTrue("gt_confidence.min" in db_indexes)
+        self.assertTrue("gt_confidence.max" in db_indexes)
+        self.assertFalse("gt_label_counts.label" in db_indexes)
+        self.assertFalse("gt_label_counts.count" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.label" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.min" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.max" in db_indexes)
+
+        self.assertTrue("frames_gt_label" in db_indexes)
+        self.assertTrue("frames_gt_confidence.min" in db_indexes)
+        self.assertTrue("frames_gt_confidence.max" in db_indexes)
+        self.assertFalse("frames_gt_label_counts.label" in db_indexes)
+        self.assertFalse("frames_gt_label_counts.count" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.label" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.min" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.max" in db_indexes)
+
+        update_fields = dataset.check_summary_fields()
+
+        self.assertListEqual(update_fields, [])
+
+        label_upper = F("label").upper()
```

Comment:
noob question - does this just change all the field names to upper case?

Reply:
yep: https://docs.voxel51.com/api/fiftyone.core.expressions.html#fiftyone.core.expressions.ViewExpression.upper